src/hash.h

```diff
@@ -132,15 +132,17 @@ class CHashWriter
 {
 private:
     CHash256 ctx;
+    size_t nBytesHashed;
```

Comment:
I would argue against adding this. We should move toward getting rid of this kind of rule rather than working toward helping adding more.

Also, the implementation is poor because ctx already track the amount of byte hashed - it is required as part of the finalization step of sha256.